### DISCUSSION TITLE
SQL issue towards QueryBuilder

### DISCUSSION THREAD

**Discussion by bglevon** [doctrine/doctrine2#9105] [Support Questions]
Hello,

I need your help, I have an SQL query that I need to change to QueryBuilder, but I can't seem to change this statement.

I've tried many ways, but my solutions don't work. 

`LEFT JOIN page p ON (p.fk_module_id = mm.id_menu_module OR p.fk_sous_module_id = msm.id_menu_sous_module)`

Thanks in advance for your help.

**Comment by leberknecht** [comment]
Not sure if it helps, but "join with conditions" you can do with something like
```php
->join('o.page', 'p', Expr\Join::WITH, 'p.fkModule = :module OR p.fkSousModule = :sousModule')
->setParameter('module', $module)
->setParameter('sousModule', $sousModule)
```
..assuming that you have setup your entities etc..

**Reply by bglevon** [reply]
Hello,
Thanks for your help. I will test this.

Regards,
